src/rest/RequestHandler.js

```diff
@@ -47,14 +55,35 @@ class RequestHandler {
     return this.execute(this.queue.shift());
   }
 
+  get globalLimited() {
+    return this.manager.globalRemaining <= 0 && Date.now() < this.manager.globalReset;
+  }
+
+  get localLimited() {
+    return this.remaining <= 0 && Date.now() < this.reset;
+  }
+
   get limited() {
-    return Boolean(this.manager.globalTimeout) || (this.remaining <= 0 && Date.now() < this.reset);
+    return this.globalLimited || this.localLimited;
   }
 
   get _inactive() {
     return this.queue.length === 0 && !this.limited && this.busy !== true;
   }
 
+  globalDelayFor(ms) {
+    return new Promise(resolve => {
+      this.manager.client.setTimeout(
+        manager => {
+          manager.globalDelay = null;
+          resolve();
+        },
+        ms,
+        this.manager,
```

Comment:
We do not pass any additional parameters to callbacks anywhere else.
```suggestion
        () => {
          this.manager.globalDelay = null;
          resolve();
        },
        ms,
```

Reply:
Also standardized the formatting with what is used for other callbacks:
```js
      this.manager.client.setTimeout(() => {
        this.manager.globalDelay = null;
        resolve();
      }, ms);
```